plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/data/OracleTimestampConverter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.oracle.data;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.ext.oracle.model.OracleConstants;
+
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.Timestamp;
+import java.util.Calendar;
+
+public class OracleTimestampConverter {
+    // See https://docs.oracle.com/en/database/oracle/oracle-database/12.2/jajdb/oracle/sql/TIMESTAMP.html#timestampValue--
+    private static final String TO_TIMESTAMP_METHOD_NAME = "timestampValue";
+
+    public static Timestamp toTimestamp(@NotNull Object object, @NotNull Connection connection) throws Exception {
+        Class<?> aClass = object.getClass();
+        switch (aClass.getName()) {
+            case OracleConstants.TIMESTAMP_CLASS_NAME:
+                return (Timestamp) getNativeMethod(aClass, TO_TIMESTAMP_METHOD_NAME)
+                        .invoke(object);
+            case OracleConstants.TIMESTAMPTZ_CLASS_NAME:
+                return (Timestamp) getNativeMethod(aClass, TO_TIMESTAMP_METHOD_NAME, Connection.class)
+                        .invoke(object, connection);
+            case OracleConstants.TIMESTAMPLTZ_CLASS_NAME:
+                return (Timestamp) getNativeMethod(aClass, TO_TIMESTAMP_METHOD_NAME, Connection.class, Calendar.class)
+                        .invoke(object, connection, Calendar.getInstance());
+        }
+        throw new DBException("Unsupported Oracle TIMESTAMP type: " + aClass.getName());
+    }
+
+    private static Method getNativeMethod(Class<?> aClass, String name, Class<?> ... args) throws NoSuchMethodException {
+        Method method = aClass.getMethod(name, args);
+        method.setAccessible(true);
+        return method;
+    }
```

Comment:
Please use `org.jkiss.utils.BeanUtils#invokeObjectMethod` instead

Reply:
Thank you for careful review.
I'm just newbie as Dbeaver contributor, So your review means a lot to me.